azure-devops/azext_devops/dev/team/team.py

```diff
@@ -0,0 +1,158 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+from vsts.core.v4_0.models.web_api_team import WebApiTeam
+from vsts.exceptions import VstsServiceError
+from azext_devops.dev.common.services import (get_core_client,
+                                              resolve_instance_and_project)
+
+
+def create_team(name, description=None, devops_organization=None, project=None, detect=None):
+    """Create a team.
+    :param name: Name of the new team.
+    :type name: str
+    :param description: Description of the new team.
+    :type description: str
+    :param devops_organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
+    :type devops_organization: str
+    :param project: Name or ID of the project.
+    :type project: str
+    :param detect: When 'On' unsupplied arg values will be detected from the current working
+                   directory's repo.
+    :type detect: str
+    :rtype: :class:`<WebApiTeam> <core.v4_0.models.WebApiTeam>`
+    """
+    try:
+        devops_organization, project = resolve_instance_and_project(detect=detect,
+                                                                    devops_organization=devops_organization,
+                                                                    project=project)
+        core_client = get_core_client(devops_organization)
+        team_to_create = WebApiTeam(name=name, description=description)
+        return core_client.create_team(team=team_to_create, project_id=project)
+    except VstsServiceError as ex:
+        raise CLIError(ex)
+
+
+def delete_team(team_id, devops_organization=None, project=None, detect=None):
+    """Deletes a team.
+    :param team_id: The name or id of the team to delete.
```

Comment:
nit: should it be called _--team_ or _--team-id_, _--id_ since it accepts both name and id?
For project and repo we have like that.

Reply:
Yeah makes sense. My only concern is then commands would look a little odd (with team word twice in command instead of just id) , which is not the case at other places. Since project or repo name/id is passed as an optional or auto-detect argument in those commands.

For example : `az devops team show --team "XYZ Team" ` or
`az devops team delete --team "XYZ Team" `

@gauravsaralMs  , @geverghe  , @snankani . Thoughts?

Reply:
I would suggest name and id; isn't name id always scoped to the command object that is being referred to? Is there any other universal name and ID? The only two universal concepts seem to be project and organization which are both referred to as --project --organization. Therefore, my suggestion would az devops team --name "XYZ Team" / az devops team --id 123

Reply:
@geverghe  Are you suggesting to split the argument ? 
Actually a single argument would suffice here. Either name or id. Because API accepts either of them. There is no universal name and ID.  What Atul referred is done in artifacts [for feed argument](https://github.com/Microsoft/vsts-cli/blob/45242f0d300a94549ed5c58ebf2458bf4fdb0ba6/azure-devops/azext_devops/dev/artifacts/universal.py#L22)